dra-evolution/pkg/api/claim_types.go

```diff
@@ -286,41 +343,29 @@ type AllocationResult struct {
 	// +optional
 	DriverData []DriverData `json:"driverData,omitempty" protobuf:"bytes,1,opt,name=driverData"`
 
-	// This is the node which provides the allocated devices.
+	// Setting this field is optional. If unset, the allocated devices are available everywhere.
 	//
-	// If empty, then the devices are not local to one particular node.
-	NodeName string `json:"nodeName,omitempty" protobuf:"bytes,2,opt,name=nodeName"`
+	// +optional
+	AvailableOnNodes *v1.NodeSelector `json:"availableOnNodes,omitempty" protobuf:"bytes,2,opt,name=availableOnNodes"`
 
-	// Shareable determines whether the resource supports more
-	// than one consumer at a time.
+	// Shareable determines whether the claim supports more than one consumer at a time.
+	//
 	// +optional
 	Shareable bool `json:"shareable,omitempty" protobuf:"varint,3,opt,name=shareable"`
 }
 
 // DriverData holds information for processing by a specific kubelet plugin.
 type DriverData struct {
-	// DriverName specifies the name of the resource driver whose kubelet
+	// DriverName specifies the name of the DRA driver whose kubelet
 	// plugin should be invoked to process the allocation once the claim is
 	// needed on a node.
 	//
 	// Must be a DNS subdomain and should end with a DNS domain owned by the
 	// vendor of the driver.
 	DriverName string `json:"driverName" protobuf:"bytes,1,name=driverName"`
 
-	// Data contains all information about the allocation that the kubelet
-	// plugin will need.
-	//
-	// +optional
-	Data *StructuredDriverData `json:"data,omitempty" protobuf:"bytes,2,opt,name=data"`
-
-	// Alternative to "Data", not describe further here for the sake of brevity:
-	// OpaqueData *string // Set by control plane controller when using "classic DRA".
```

Comment:
I came to the conclusion that we don't need this opaque data and thus `StructuredDriverData` can be inlined.

A control plane controller can and should use instead the in-tree RequestAllocationResult. Then selecting a specific request for a container works the same way in both allocation modes.

Reply:
So if I'm reading this diff correctly, the only fields in this struct now are `DriverName` and `Config`? Is this because the new `Config` section is (under the hood) just a RawExtension, so that can now be used to encapsulate any opaque data that needs to be passed through?

Reply:
Yes, that was the idea.